dataline-server/src/main/java/io/dataline/server/apis/ConfigurationApi.java

```diff
@@ -116,177 +121,194 @@ public ConfigurationApi(final Path dbRoot, BasicDataSource connectionPool) {
-    return connectionsHandler.updateConnection(connectionUpdate);
+    return execute(() -> connectionsHandler.updateConnection(connectionUpdate));
   }
 
   @Override
   public ConnectionReadList listConnectionsForWorkspace(@Valid WorkspaceIdRequestBody workspaceIdRequestBody) {
-    return connectionsHandler.listConnectionsForWorkspace(workspaceIdRequestBody);
+    return execute(() -> connectionsHandler.listConnectionsForWorkspace(workspaceIdRequestBody));
   }
 
   @Override
   public ConnectionRead getConnection(@Valid ConnectionIdRequestBody connectionIdRequestBody) {
-    return connectionsHandler.getConnection(connectionIdRequestBody);
+    return execute(() -> connectionsHandler.getConnection(connectionIdRequestBody));
   }
 
   @Override
   public ConnectionSyncRead syncConnection(@Valid ConnectionIdRequestBody connectionIdRequestBody) {
-    return schedulerHandler.syncConnection(connectionIdRequestBody);
+    return execute(() -> schedulerHandler.syncConnection(connectionIdRequestBody));
   }
 
   // JOB HISTORY
 
   @Override
   public JobReadList listJobsFor(@Valid JobListRequestBody jobListRequestBody) {
-    return jobHistoryHandler.listJobsFor(jobListRequestBody);
+    return execute(() -> jobHistoryHandler.listJobsFor(jobListRequestBody));
   }
 
   @Override
   public JobInfoRead getJobInfo(@Valid JobIdRequestBody jobIdRequestBody) {
-    return jobHistoryHandler.getJobInfo(jobIdRequestBody);
+    return execute(() -> jobHistoryHandler.getJobInfo(jobIdRequestBody));
   }
 
   // WEB BACKEND
 
   @Override
   public WbConnectionReadList webBackendListConnectionsForWorkspace(@Valid WorkspaceIdRequestBody workspaceIdRequestBody) {
-    return webBackendConnectionsHandler.webBackendListConnectionsForWorkspace(workspaceIdRequestBody);
+    return execute(() -> webBackendConnectionsHandler.webBackendListConnectionsForWorkspace(workspaceIdRequestBody));
   }
 
   @Override
   public WbConnectionRead webBackendGetConnection(@Valid ConnectionIdRequestBody connectionIdRequestBody) {
-    return webBackendConnectionsHandler.webBackendGetConnection(connectionIdRequestBody);
+    return execute(() -> webBackendConnectionsHandler.webBackendGetConnection(connectionIdRequestBody));
+  }
+
+  private <T> T execute(HandlerCall<T> call) {
+    try {
+      return call.call();
+    } catch (ConfigNotFoundException e) {
+      throw new KnownException(
+          HttpStatus.NOT_FOUND_404,
+          String.format("Could not find configuration for %s: %s.", e.getType().toString(), e.getConfigId()), e);
+    } catch (JsonValidationException e) {
+      throw new KnownException(
+          HttpStatus.UNPROCESSABLE_ENTITY_422,
+          String.format("The provided configuration does not fulfill the specification. Errors: %s", e.getMessage()), e);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private interface HandlerCall<T> {
+
+    T call() throws ConfigNotFoundException, IOException, JsonValidationException;
```

Comment:
how does this scale with more exceptions? do we just keep adding to this method signature? or do we have different handlers for different api calls.

Reply:
I am thinking that for now we add more exceptions.